### PR TITLE
Update dupe check

### PR DESCRIPTION
**Pull request by dshanske**
This updates the dupe check parameters to search for dupes by matching on webmention_source_url, url, _crossposting_link or source_url in one query instead of multiple queries. 

webmention_source url = the original source url
url = the canonical url in the event of something like a Bridgy post
_crossposting_link for Salmon
source_url for ActivityPub

### CODE REVIEW

**Comment by dshanske** [comment]
Makes me wonder if we shouldn't migrate to source_url to match activitypub.

File: includes/class-receiver.php [hunk 1]
```diff
@@ -501,10 +501,7 @@ public static function disable_wp_check_dupes( $dupe_id, $commentdata ) {
 			return $dupe_id;
 		}
 
-		if (
-			( isset( $commentdata['comment_type'] ) && 'webmention' === $commentdata['comment_type'] ) ||
-			( isset( $commentdata['comment_meta'] ) && ! empty( $commentdata['comment_meta']['semantic_linkbacks_type'] ) )
-		) {
+		if ( ! empty( $commentdata['comment_meta']['protocol'] ) && 'webmention' === $commentdata['comment_meta']['protocol'] ) {
```
Comment (pfefferle) [inline includes/class-receiver.php:504]: can we alway assume that the array-keys exist?

Comment (dshanske) [reply includes/class-receiver.php:504]: This is a check that is only supposed to be triggered on a Webmention, but no issue with an existence check

File: includes/class-receiver.php [hunk 2]
```diff
@@ -524,28 +521,50 @@ public static function check_dupes( $commentdata ) {
...
-			$args = array(
-				'post_id'    => $commentdata['comment_post_ID'],
-				'meta_key'   => 'webmention_source_url',
-				'meta_value' => $commentdata['comment_author_url'],
+			),
+		);
+
+		if ( ! empty( $fragment ) ) {
+			// Ensure that if there is a
+			$args['meta_query'][0][] = array(
```
Comment (pfefferle) [inline includes/class-receiver.php:564]: this looks a bit tricky, if the array order changes for example. can we make this a bit more robust?

File: includes/class-receiver.php (hunk 1 above)
Comment (pfefferle) [reply includes/class-receiver.php:504]: yes, sorry. wasn't aware, that `empty` also checks for `isset`.

**Comment by dshanske** [comment]
I don't think we do. Don't we have to decode them?

**Comment by snarfed** [comment]
for reference: https://github.com/pfefferle/wordpress-webmention/issues/359#issuecomment-1498347387

**Comment by pfefferle** [comment]
@dshanske not decode, but encode! Because they are saved encoded didn't they? `esc_url`??

**Comment by dshanske** [comment]
This is how I confuse myself

**Comment by pfefferle** [comment]
😊

I think we should add some tests, to see what is correct!

**Comment by pfefferle** [comment]
I would love to use this, to add some tests. Maybe @snarfed has some "special" URLs for us ;)

**Comment by dshanske** [comment]
@pfefferle Do you want to merge this and do the urlencode testing as a separate PR?

**Review by pfefferle — APPROVED** [review]
LGTM

**Pull request merged**